### PR TITLE
DP-23508: Upgrade to PHP 8

### PR DESCRIPTION
**Pull request by arthurbaghdas**
**Description:**
Reroll:

```console
ddev composer upgrade -W \
  drupal/diff \
  drupal/acquia_connector \
  drupal/geocoder \
  drupal/geofield \
  drupal/memcache \
  drupal/monolog \
  palantirnet/drupal-rector \
  geocoder-php/open-cage-provider \
  monolog/monolog \
  typhonius/acquia-php-sdk-v2 \
  chillerlan/php-qrcode
```

**Jira:** (Skip unless you are MA staff)
DP-23508

**To Test:**
- [ ] Add steps to test this feature

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -375,9 +375,6 @@
                 "CKeditor Plugin - using a combined patch to avoid conflict (https://www.drupal.org/node/2928318)": "https://www.drupal.org/files/issues/2020-10-28/media_entity_download-ckeditor_plugin-2928318-20.patch",
                 "3106062 - Track the usage of Media entity through media_entity_download route with entity_usage": "patches/media_entity_download-track_media_entity_usage-3106062-2.patch"
             },
-            "drupal/memcache": {
-                "Split large (> 1mb) items (https://www.drupal.org/node/435694)": "https://www.drupal.org/files/issues/2019-02-26/435694-118-d8-memcache-large-items.patch"
-            },
```
Comment (weitzman) [inline composer.json:399]: Is this patch no longer needed on PHP8?

Comment (arthurbaghdas) [reply composer.json:399]: Its already in the latest version of the module. Which I've updated

**Review by weitzman — APPROVED** [review]

**Review by weitzman — CHANGES_REQUESTED** [review]
LGTM except for https://github.com/massgov/openmass/blob/5cdc0f4c0c7ed6bdddc740d03ddbf46f1108c91b/composer.json#L16 needs changing, and thus a new composer.lock I think.

**Comment by deviantintegral** [comment]
Looks like the reroll instructions are no longer valid:

```console
$ ddev composer upgrade drupal/diff drupal/acquia_connector drupal/geocoder drupal/geofield drupal/memcache palantirnet/drupal-rector drupal/monolog
Project-level command 'mysql' is overriding the global 'mysql' command
Project-level command 'launch' is overriding the global 'launch' command
Project-level command 'xdebug' is overriding the global 'xdebug' command
Gathering patches for root package.
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - geocoder-php/open-cage-provider is locked to version 4.4.0 and an update of this package was not requested.
    - geocoder-php/open-cage-provider 4.4.0 requires php ^7.2 -> your php version (8.0; overridden via config.platform, actual: 7.4.26) does not satisfy that requirement.
  Problem 2
    - chillerlan/php-qrcode 3.4.0 requires php ^7.2 -> your php version (8.0; overridden via config.platform, actual: 7.4.26) does not satisfy that requirement.
    - drupal/ga_login 1.0.0-alpha6 requires chillerlan/php-qrcode ^1.0.9 || ^2.0 || ^3.2 || ^4.0 -> satisfiable by chillerlan/php-qrcode[3.4.0].
    - drupal/ga_login is locked to version 1.0.0-alpha6 and an update of this package was not requested.

composer [upgrade drupal/diff drupal/acquia_connector drupal/geocoder drupal/geofield drupal/memcache palantirnet/drupal-rector drupal/monolog] failed, composer command failed: exit status 2. stderr=
```

**Comment by deviantintegral** [comment]
I used the following to get the branch updated:

```
ddev composer upgrade -w drupal/diff drupal/acquia_connector drupal/geocoder drupal/geofield drupal/memcache palantirnet/drupal-rector drupal/monolog geocoder-php/open-cage-provider drupal/ga_login typhonius/acquia-php-sdk-v2
```

I've merged and pushed, but haven't tested _at all_ yet. Tomorrow!

**Comment by deviantintegral** [comment]
The test failure only happens on the first request, which gives us a `cache-control: must-revalidate, no-cache, private` response. The second request (even after a `drush cr`) works fine.

I've confirmed this happens locally running this branch, but downgrading back to PHP 7.4. Since it's a _long_ time to reset locally (ddev delete, plus ddev start doing it's initial copy of the database volume, plus drush deploy) I'm going to try and isolate which change in this PR triggers this behaviour to limit the code to investigate.

**Comment by deviantintegral** [comment]
I've found the problem! https://git.drupalcode.org/project/akamai/-/blob/4.x/src/AkamaiAuthentication.php#L56

What's odd is I would expect this to happen after every cache clear, but I can only replicate it once after running deploy. Assuming `composer install` has been run on this branch, run:

```console
sudo ls; ddev delete -Oy; ddev start; ddev drush deploy; ddev exec phpunit --filter testEmergencyAlertResponseSitewide
```

The test should fail. Run the above again after commenting out the above line, and it should pass.

**Comment by deviantintegral** [comment]
I discovered that I could replicate by running `drush deploy`, and with a bit of trial and error, it looks like it's the `config:import` that does _something_ that replicates this:

```console
ddev drush -y cim && ddev exec phpunit --filter testEmergencyAlertResponseSitewide
+------------+---------------------------+-----------+
| Collection | Config                    | Operation |
+------------+---------------------------+-----------+
|            | acquia_connector.settings | Update    |
+------------+---------------------------+-----------+

 // Import the listed configuration changes?: yes.

 [notice] Synchronized configuration: update acquia_connector.settings.
 [notice] Finalizing configuration synchronization.
 [success] The configuration was imported successfully.
PHPUnit 9.5.19 #StandWithUkraine

F                                                                   1 / 1 (100%)

Time: 00:03.459, Memory: 70.00 MB

There was 1 failure:

1) Drupal\Tests\mass_alerts\ExistingSite\EmergencyAlertsTest::testEmergencyAlertResponseSitewide
Failed asserting that 'must-revalidate, no-cache, private' contains "stale-if-error=604800".

/var/www/html/vendor/phpunit/phpunit/src/Framework/Constraint/Constraint.php:121
/var/www/html/vendor/phpunit/phpunit/src/Framework/Constraint/Constraint.php:55
/var/www/html/docroot/modules/custom/mass_alerts/tests/src/ExistingSite/EmergencyAlertsTest.php:89
/var/www/html/vendor/phpunit/phpunit/src/Framework/TestResult.php:726
/var/www/html/vendor/phpunit/phpunit/src/Framework/TestSuite.php:677
/var/www/html/vendor/phpunit/phpunit/src/Framework/TestSuite.php:677
/var/www/html/vendor/phpunit/phpunit/src/Framework/TestSuite.php:677
/var/www/html/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:673
/var/www/html/vendor/phpunit/phpunit/src/TextUI/Command.php:143
/var/www/html/vendor/phpunit/phpunit/src/TextUI/Command.php:96

FAILURES!
Tests: 1, Assertions: 2, Failures: 1.
Failed to execute command phpunit --filter testEmergencyAlertResponseSitewide: exit status 1
```

I'm not sure why the config:diff test wasn't catching that (perhaps we have an environment variable set that causes `REDACTED` value to be in config?), but let's see if this passes in CI as it does for me locally.

Even if it does pass in CI, it seems like the Akamai module should be logging that (with a better message) instead of using messenger.

**Comment by deviantintegral** [comment]
I reverted the logging changes with https://github.com/massgov/openmass/pull/1286/commits/787838e241444cd2b44cf2cad44e10807f42e9c3 as noted at https://github.com/massgov/openmass/pull/1427.

**Comment by deviantintegral** [comment]
It looks like there still PHP 7 containers or code in the mix. On trying to deploy, Circle doesn't get past trying to SSH to Acquia:

```
#!/bin/bash -eo pipefail
drush ma:deploy -v -y  --refresh-db test feature/DP-23508_upgrade_to_php8
 [info] Starting bootstrap to none
 [info] Drush bootstrap phase 0
 [info] Try to validate bootstrap phase 0
 [success] Starting deployment of feature/DP-23508_upgrade_to_php8 to test at 2022-03-31 1:44:57 PM
 [info] Executing: /tmp/code/vendor/drush/drush/drush ma:latest-backup-url prod --uri=default --root=/tmp/code/docroot
 [success] Backup URL retrieved.
 [info] Executing: ssh -o ForwardAgent=yes -o ServerAliveInterval=30 *******.test@*******stg.ssh.prod.acquia-sites.com 'wget -q --continue '\''https://*******.prod.acquia-sites.com/AH_DOWNLOAD?t=1648748702&d=/mnt/gfs/*******.prod/backups/prod-*******-*******-2022-03-31.sql.gz&prod=9bdc41cdc1da5a23c642420fd90a59ae3929850d2c825ae25a9598e520cd1390'\'' --output-document=/mnt/tmp/1648748697-db-backup.sql.gz'
> Warning: Permanently added '*******stg.ssh.prod.acquia-sites.com,35.171.11.198' (RSA) to the list of known hosts.
 [success] Database downloaded from backup.
 [info] Executing: ssh -o ForwardAgent=yes -o ServerAliveInterval=30 *******.test@*******stg.ssh.prod.acquia-sites.com 'drush sql:drop --uri=https://edit.stage.mass.gov --root=/var/www/html/*******.test/docroot'

In SiteProcess.php line 214:
                                                                               
  [Symfony\Component\Process\Exception\ProcessFailedException]                 
  The command "ssh -o ForwardAgent=yes -o ServerAliveInterval=30 *******.test  
  @*******stg.ssh.prod.acquia-sites.com 'drush sql:drop --uri=https://edit.st  
  age.mass.gov --root=/var/www/html/*******.test/docroot'" failed.             
                                                                               
  Exit Code: 255(Unknown error)                                                
                                                                               
  Working directory:                                                           
                                                                               
  Output:                                                                      
  ================                                                             
                                                                               
  Parse error: syntax error, unexpected token "match" in /mnt/www/html/massgo  
  vstg/vendor/caxy/php-htmldiff/lib/Caxy/HtmlDiff/HtmlDiff.php on line 699     
                                                                               
                                                                               
  Error Output:                                                                
  ================                                                             
                                                                               

Exception trace:
  at /tmp/code/vendor/consolidation/site-process/src/SiteProcess.php:214
 Consolidation\SiteProcess\SiteProcess->mustRun() at /tmp/code/drush/Commands/DeployCommands.php:298
 Drush\Commands\DeployCommands->deploy() at n/a:n/a
 call_user_func_array() at /tmp/code/vendor/consolidation/annotated-command/src/CommandProcessor.php:257
 Consolidation\AnnotatedCommand\CommandProcessor->runCommandCallback() at /tmp/code/vendor/consolidation/annotated-command/src/CommandProcessor.php:212
 Consolidation\AnnotatedCommand\CommandProcessor->validateRunAndAlter() at /tmp/code/vendor/consolidation/annotated-command/src/CommandProcessor.php:176
 Consolidation\AnnotatedCommand\CommandProcessor->process() at /tmp/code/vendor/consolidation/annotated-command/src/AnnotatedCommand.php:350
 Consolidation\AnnotatedCommand\AnnotatedCommand->execute() at /tmp/code/vendor/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /tmp/code/vendor/symfony/console/Application.php:1027
 Symfony\Component\Console\Application->doRunCommand() at /tmp/code/vendor/symfony/console/Application.php:273
 Symfony\Component\Console\Application->doRun() at /tmp/code/vendor/symfony/console/Application.php:149
 Symfony\Component\Console\Application->run() at /tmp/code/vendor/drush/drush/src/Runtime/Runtime.php:121
 Drush\Runtime\Runtime->doRun() at /tmp/code/vendor/drush/drush/src/Runtime/Runtime.php:51
 Drush\Runtime\Runtime->run() at /tmp/code/vendor/drush/drush/drush.php:72
 require() at /tmp/code/vendor/drush/drush/drush:4


Exited with code exit status 1
CircleCI received exit code 1
```

**Comment by deviantintegral** [comment]
`composer install` shows very different package changes compared to my local:

```
#!/bin/bash -eo pipefail
composer install --no-interaction --optimize-autoloader
Gathering patches for root package.
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Package operations: 8 installs, 0 updates, 0 removals
  - Downloading ckeditor/liststyle (4.11.1)
  - Downloading components/highlightjs (9.7.0)
  - Downloading d3/d3 (v3.5.17)
  - Downloading desandro/imagesloaded (v4.1.1)
  - Downloading desandro/masonry (4.1.1)
  - Downloading enyo/dropzone (5.7.2)
  - Downloading furf/jquery-ui-touch-punch (dev-master)
  - Downloading *******/mayflower-artifacts (dev-develop 0903a92)
Gathering patches for root package.
Gathering patches for dependencies. This might take a minute.
  - Installing ckeditor/liststyle (4.11.1): Extracting archive
  - Installing components/highlightjs (9.7.0): Extracting archive
  - Installing d3/d3 (v3.5.17): Extracting archive
  - Installing desandro/imagesloaded (v4.1.1): Extracting archive
  - Installing desandro/masonry (4.1.1): Extracting archive
  - Installing enyo/dropzone (5.7.2): Extracting archive
  - Installing furf/jquery-ui-touch-punch (dev-master): Extracting archive
  - Installing *******/mayflower-artifacts (dev-develop 0903a92): Extracting archive
  - Applying patches for ckeditor/liststyle
    https://www.drupal.org/files/issues/2019-01-02/ckeditor_liststyle-styles-to-type-2874952-4-do-not-test.patch (Change code from styles to type)

Package doctrine/reflection is abandoned, you should avoid using it. Use roave/better-reflection instead.
Package mandrill/mandrill is abandoned, you should avoid using it. Use mailchimp/transactional instead.
Package webmozart/path-util is abandoned, you should avoid using it. Use symfony/filesystem instead.
Generating optimized autoload files
Class Mandrill_Error located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_HttpError located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_ValidationError located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Invalid_Key located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_PaymentRequired located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_Subaccount located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_Template located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_ServiceUnavailable located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_Message located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Invalid_Tag_Name located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Invalid_Reject located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_Sender located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_Url located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_TrackingDomain located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Invalid_Template located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_Webhook located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_InboundDomain located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_InboundRoute located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_Export located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_IP_ProvisionLimit located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_Pool located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_NoSendingHistory located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_PoorReputation located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_IP located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Invalid_EmptyDefaultPool located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Invalid_DeleteDefaultPool located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Invalid_DeleteNonEmptyPool located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Invalid_CustomDNS located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Invalid_CustomDNSPending located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Metadata_FieldLimit located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
Class Mandrill_Unknown_MetadataField located in ./vendor/mandrill/mandrill/src/Mandrill/Exceptions.php does not comply with psr-0 autoloading standard. Skipping.
105 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
Scaffolding files for drupal/core:
  - Skip [web-root]/.eslintignore: overridden in mass/*******
  - Skip [web-root]/.eslintrc.json: overridden in mass/*******
  - Skip [web-root]/.ht.router.php: overridden in mass/*******
  - Skip [web-root]/.htaccess: overridden in mass/*******
  - Skip [web-root]/example.gitignore: overridden in mass/*******
  - Skip [web-root]/INSTALL.txt: overridden in mass/*******
  - Skip [web-root]/robots.txt: overridden in mass/*******
  - Skip [web-root]/sites/README.txt: overridden in mass/*******
  - Copy [web-root]/sites/development.services.yml from assets/scaffold/files/development.services.yml
  - Skip [web-root]/modules/README.txt: overridden in mass/*******
  - Skip [web-root]/profiles/README.txt: overridden in mass/*******
  - Skip [web-root]/themes/README.txt: overridden in mass/*******
CircleCI received exit code 0
```

Instead of:

```
~/.composer/vendor/bin/composer-lock-diff --from origin/develop
+-----------------------------------+-------------+---------+----------------------------------------------------------------------------------+
| Production Changes                | From        | To      | Compare                                                                          |
+-----------------------------------+-------------+---------+----------------------------------------------------------------------------------+
| behat/mink                        | v1.9.0      | v1.10.0 | https://github.com/minkphp/Mink/compare/v1.9.0...v1.10.0                         |
| caxy/php-htmldiff                 | v0.1.5      | v0.1.14 | https://github.com/caxy/php-htmldiff/compare/v0.1.5...v0.1.14                    |
| chillerlan/php-qrcode             | 3.4.0       | 4.3.3   | https://github.com/chillerlan/php-qrcode/compare/3.4.0...4.3.3                   |
| chillerlan/php-settings-container | 1.2.1       | 2.1.3   | https://github.com/chillerlan/php-settings-container/compare/1.2.1...2.1.3       |
| clue/stream-filter                | v1.4.1      | v1.6.0  | https://github.com/clue/stream-filter/compare/v1.4.1...v1.6.0                    |
| drupal/acquia_connector           | 2.0.0-rc1   | 2.0.0   | https://git.drupalcode.org/project/acquia_connector/compare/2.0.0-rc1...2.0.0    |
| drupal/diff                       | e7f5605     | 1.0.0   | https://git.drupalcode.org/project/diff/compare/e7f5605...1.0.0                  |
| drupal/environment_indicator      | 4.0.5       | REMOVED |                                                                                  |
| drupal/geocoder                   | 3.8.0       | 3.25.0  | https://git.drupalcode.org/project/geocoder/compare/3.8.0...3.25.0               |
| drupal/geofield                   | 1.15.0      | 1.38.0  | https://git.drupalcode.org/project/geofield/compare/1.15.0...1.38.0              |
| drupal/memcache                   | 2.1.0       | 2.3.0   | https://git.drupalcode.org/project/memcache/compare/2.1.0...2.3.0                |
| drupal/monolog                    | 2.0.0-beta2 | 2.0.1   | https://git.drupalcode.org/project/monolog/compare/2.0.0-beta2...2.0.1           |
| drupal/scheduled_transitions      | 2.2.1       | 1b3d8c5 | https://git.drupalcode.org/project/scheduled_transitions/compare/2.2.1...1b3d8c5 |
| ezyang/htmlpurifier               | v4.9.3      | v4.14.0 | https://github.com/ezyang/htmlpurifier/compare/v4.9.3...v4.14.0                  |
| geocoder-php/common-http          | 4.3.0       | 4.4.0   | https://github.com/geocoder-php/php-common-http/compare/4.3.0...4.4.0            |
| geocoder-php/open-cage-provider   | 4.4.0       | 4.5.0   | https://github.com/geocoder-php/open-cage-provider/compare/4.4.0...4.5.0         |
| jawira/case-converter             | v1.2.0      | REMOVED |                                                                                  |
| league/oauth2-client              | 2.4.1       | 2.6.1   | https://github.com/thephpleague/oauth2-client/compare/2.4.1...2.6.1              |
| massgov/mayflower-artifacts       | de1f7f1     | 0903a92 | https://github.com/massgov/mayflower-artifacts/compare/de1f7f1...0903a92         |
| monolog/monolog                   | 2.1.1       | 2.4.0   | https://github.com/Seldaek/monolog/compare/2.1.1...2.4.0                         |
| palantirnet/drupal-rector         | 0.5.6       | 0.12.1  | https://github.com/palantirnet/drupal-rector/compare/0.5.6...0.12.1              |
| paragonie/constant_time_encoding  | v2.4.0      | v2.5.0  | https://github.com/paragonie/constant_time_encoding/compare/v2.4.0...v2.5.0      |
| php-http/discovery                | 1.9.1       | 1.14.1  | https://github.com/php-http/discovery/compare/1.9.1...1.14.1                     |
| php-http/guzzle6-adapter          | v1.1.1      | v2.0.2  | https://github.com/php-http/guzzle6-adapter/compare/v1.1.1...v2.0.2              |
| php-http/httplug                  | v1.1.0      | 2.3.0   | https://github.com/php-http/httplug/compare/v1.1.0...2.3.0                       |
| php-http/message                  | 1.8.0       | 1.13.0  | https://github.com/php-http/message/compare/1.8.0...1.13.0                       |
| rector/rector-prefixed            | v0.7.29     | REMOVED |                                                                                  |
| sunra/php-simple-html-dom-parser  | v1.5.2      | REMOVED |                                                                                  |
| willdurand/geocoder               | 4.3.0       | 4.5.0   | https://github.com/geocoder-php/php-common/compare/4.3.0...4.5.0                 |
| phpstan/phpstan                   | NEW         | 1.5.1   |                                                                                  |
| psr/http-client                   | NEW         | 1.0.1   |                                                                                  |
| rector/rector                     | NEW         | 0.12.19 |                                                                                  |
+-----------------------------------+-------------+---------+----------------------------------------------------------------------------------+

+-----------------------+-------+-------+----------------------------------------------------------------+
| Dev Changes           | From  | To    | Compare                                                        |
+-----------------------+-------+-------+----------------------------------------------------------------+
| phpstan/phpdoc-parser | 1.2.0 | 1.4.0 | https://github.com/phpstan/phpdoc-parser/compare/1.2.0...1.4.0 |
+-----------------------+-------+-------+----------------------------------------------------------------+
```

My guess is it's an issue with caching the vendor directory, and it wouldn't surprise me if composer patches is making things trickier as well. On my local, I have to re-run composer install 3-4 times for it to work (though I never get output similar to the above).

I will likely pick this up again tomorrow.

File: changelogs/DP-23508-5.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Changed:
```
Comment (weitzman) [inline changelogs/DP-23508-5.yml:1]: I havent seen a filename with a dash after the issue number. Seems like its supported so OK.

File: composer.json [hunk 3]
```diff
@@ -350,19 +350,16 @@
...
+                "Enable entity_hierarchy_breadcrumb module to work on node edit routes": "patches/entity_hierarchy_breadcrumb_for_admin_routes.patch"
             },
             "drupal/entity_usage": {
                 "2954973 - Track absolute URLs in link fields": "https://www.drupal.org/files/issues/2021-08-17/2954973-63.patch",
                 "3007752 - Entity usage list does not scale (https://www.drupal.org/project/entity_usage/issues/3007752)": "patches/entity_usage-usage_list_scale-3007752-17.patch",
                 "Hides language column and shows bundle" : "patches/entity-usage--hide-language--show-bundle.patch",
                 "3254437 - Delete queue items for `entity_usage_regenerate_queue` when `running drush entity-usage:recreate`  ": "https://www.drupal.org/files/issues/2021-12-14/issue-3254437--remove-items-from-queue-before-recreating-it.patch",
-                "3015287 - Allow usage records to be registered in background" : "https://www.drupal.org/files/issues/2019-07-22/3015287-12.patch",
```
Comment (weitzman) [inline composer.json]: @beto-aveiga This change looks OK to you?

**Review by weitzman — COMMENTED** [review]
did we lose the code to switch php version at acquia during a deploy? i am not seeing it.

File: composer.json (hunk 3 above)
Comment (beto-aveiga) [reply composer.json]: @weitzman no, that patch is needed to make entity usage use a queue instead of process items immediately.

Comment (weitzman) [reply composer.json]: @deviantintegral See comment above ... You can ignore my comment about the missing API call to switch PHP version. I forgot that went in separately.

File: changelogs/DP-23508-5.yml (hunk 2 above)
Comment (deviantintegral) [reply changelogs/DP-23508-5.yml:1]: Yeah, this is actually the fifth time I've done that for this ticket so I think we're good. Though it really shows I should have just created new tickets and converted DP-23508 into an epic!

**Review by weitzman — APPROVED** [review]

**Comment by weitzman** [comment]
I resolved conflicts and pinned monolog/monolog to 2.6 because 2.7 introduces a problem https://www.drupal.org/project/monolog/issues/3284825.

**Pull request merged**